_ext/posts.rb

```diff
@@ -0,0 +1,137 @@
+            # check for a date inside the page first
+            if (page.date?)
+              page.relative_source_path =~ /^#{@path_prefix}\/(.*)\..*$/
+              date = page.date
+              unless date.kind_of? DateTime
+                if date.kind_of? String
+                  date = DateTime.parse date
+                elsif date.kind_of? Date
+                  date = DateTime.new(date.year, date.month, date.day)
+                elsif date.kind_of? Time
+                  date = date.to_datetime
+                end
+              end
+              year = date.year
+              month = sprintf( "%02d", date.month )
+              day = sprintf( "%02d", date.day )
+              page.date = date
+              slug = $1
+              if ( page.relative_source_path =~ /^#{@path_prefix}\/(20[01][0-9])-([01][0-9])-([0123][0-9])-([^.]+)\..*$/ )
+                slug = $4
+              end
+            elsif ( page.relative_source_path =~ /^#{@path_prefix}\/(20[01][0-9])-([01][0-9])-([0123][0-9])-([^.]+)\..*$/ )
+              year  = $1
+              month = $2
+              day   = $3
+              slug  = $4
+              page.date = DateTime.new( year.to_i, month.to_i, day.to_i )
+            end
+
+            # if a date was found create a post
+            if( year and month and day)
+              page.layout ||= @default_layout if @default_layout
+              page.slug ||= slug
+              context = page.create_context
+              if (@wp_compat)
+                page.output_path = "#{@path_prefix}/#{year}/#{month}/#{page.slug}.html"
+              else
+                page.output_path = "#{@path_prefix}/#{year}/#{month}/#{day}/#{page.slug}.html"
+              end
+              page.summary = extract_summary(page.content.force_encoding("UTF-8"))
+              posts << page
+            end
+          end
+        end
+
+        posts = posts.sort_by{|each| [each.date, each.sequence || 0, File.mtime( each.source_path ), each.slug ] }.reverse
+
+        last = nil
+        singular = @assign_to.to_s.singularize
+        posts.each do |e|
+          if ( last != nil )
+             e.send( "next_#{singular}=", last )
+             last.send( "previous_#{singular}=", e )
+          end
+          last = e
+          archive << e
+        end
+        site.pages.concat( archive.generate_pages( site.engine, archive_template, archive_path ) ) if (archive_template && archive_path)
+        site.send( "#{@assign_to}=", posts )
+        site.send( "#{@assign_to}_archive=", archive )
+
+      end
+
+      def extract_summary(post_content)
+        summary = post_content
+        # If the excerpt marker is present, we use it
+        if post_content.include?('<!--more-->')
+          summary = post_content.split('<!--more-->').take(1).join()
+        end
+
+        # Use preamble if present
+        preamble = Nokogiri::HTML::Document.parse(post_content).at("div#preamble")
```

Comment:
This seems to be the problem why the excerpt markers don't work. The logic below should only be used when the marker isn't present. If I change it into that though, I'm getting weird rendering issues. I suppose it's caused by this working on HTML instead of the source asciidoc, so one ends up with an incomplete <div> or so. Have you actually tried this out?

Reply:
I didn't see a specific place now where this is a problem after properly handling the extraction of the summary from the teaser role.  If there are places this is still problematic, let me know.